client/signup/navigation-link/index.jsx

```diff
@@ -22,10 +23,36 @@ const NavigationLink = React.createClass( {
 		flowName: React.PropTypes.string.isRequired,
 		positionInFlow: React.PropTypes.number,
 		previousPath: React.PropTypes.string,
-		signupProgressStore: React.PropTypes.array,
+		signupProgress: React.PropTypes.array,
 		stepName: React.PropTypes.string.isRequired
 	},
 
+	/**
+	 * Returns the previous step name, skipping over steps with the
+	 * `wasSkipped` property.
+	 *
+	 * @return {string|null} The previous step name
+	 */
+	getPreviousStepName() {
+		const { stepName, signupProgress } = this.props;
+
+		const currentStepIndex = findIndex( signupProgress, { stepName } );
+
+		let stepIndex = currentStepIndex - 1;
+
+		while ( stepIndex > -1 ) {
```

Comment:
What you think about something like that:
```js
return signupProgress.reverse().find( step => ! step.wasSkipped ).stepName;
```

It's much less efficient, but we don't have that many elements that it'll matter, however it's
easier to read imo.